src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -129,9 +142,11 @@ public void push(RowMap r) throws Exception {
 
 		ProducerRecord<String, String> record;
 		if (r instanceof DDLMap) {
-			record = new ProducerRecord<>(ddlTopic, this.ddlPartitioner.kafkaPartition(r, this.numPartitions), key, value);
+			String topic = generateTopic(this.ddlTopic, r);
+			record = new ProducerRecord<>(topic, this.ddlPartitioner.kafkaPartition(r, getNumPartitions(topic)), key, value);
```

Comment:
I'll back this out, just saw the update and thought it would be more complete, but yeah. the hardcoding doesn't play well.